pkgs/by-name/cl/clevis/package.nix

```diff
@@ -60,10 +60,15 @@ stdenv.mkDerivation (finalAttrs: {
     "man"
   ];
 
+  # Filter all src files that have the string "/bin/cat" and patch that string
+  # to an absolute path for our coreutils location in these files.
+  # The xargs command is a little bit convoluted because a simpler version would
+  # be vulnerable to code injection.
+  # The hint is a courtesy of https://unix.stackexchange.com/a/267438
   postPatch = ''
-    for f in $(find src/ -type f); do
-      grep -q "/bin/cat" "$f" && substituteInPlace "$f" \
-        --replace-fail '/bin/cat' '${lib.getExe' coreutils "cat"}' || true
+    for f in $(find src/ -type f -print0 |\
+                 xargs -0 -I@ sh -c 'grep -q "/bin/cat" "$1" && echo "$1"' sh @); do
+      substituteInPlace "$f" --replace-fail '/bin/cat' '${lib.getExe' coreutils "cat"}'
```

Comment:
Hmm how many files are we talking about? Would it be possible perhaps to maintain a list of them, and simplify this part a bit?

Reply:
> Hmm how many files are we talking about? Would it be possible perhaps to maintain a list of them, and simplify this part a bit?

BTW I found [this upstream issue](https://github.com/latchset/clevis/issues/61) and [PR](https://github.com/latchset/clevis/pull/64).. Perhaps it'd be much more elegent to use a `fetchpatch` with the revert flag?

Reply:
It would be better to fix this upstream.

Reply:
> It would be better to fix this upstream.

Could you at least mention the upstream issue and PR in a comment? Perhaps someone in the future would like to follow on that subject.

Reply:
Done.